flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -732,6 +758,18 @@ else if (current == JobStatus.RESTARTING) {
 		}
 	}
 
+	public void stop() {
+		if(jobType == JobType.STREAMING) {
+			for(ExecutionVertex ev : this.getAllExecutionVertices()) {
+				if(ev.getNumberOfInputs() == 0) { // send signal to sources only
+					ev.stop();
+				}
+			}
+		} else {
+			throw new RuntimeException("STOP is only supported by streaming jobs.");
```

Comment:
I think it is a bit hard to throw a `RuntimeException` in the case of a batch job. This effectively means that whenever you call stop on a batch job, the `JobManager` will die with a `RuntimeException`. I think it is better to simply tell the user that stop is not supported for batch jobs or to call the cancel method for batch jobs.